src/extension.ts

```diff
@@ -80,17 +82,55 @@ let decorationType: TextEditorDecorationType = window.createTextEditorDecoration
 
 const config = workspace.getConfiguration("metals");
 
+interface Activation {
+  shouldCreateNewProject: boolean;
+  status: Disposable;
+}
+
 export async function activate(context: ExtensionContext) {
   detectLaunchConfigurationChanges();
   checkServerVersion();
+  let activation: Activation = {
+    shouldCreateNewProject: false,
+    status: window.setStatusBarMessage("$(gear~spin) Activating Metals"),
+  };
+
+  context.subscriptions.push(
+    commands.registerCommand("metals.new-scala-project-activate", () => {
+      activation.shouldCreateNewProject = true;
+    })
+  );
 
   getJavaHome(workspace.getConfiguration("metals").get("javaHome"))
-    .then((javaHome) => fetchAndLaunchMetals(context, javaHome))
+    .then((javaHome) => fetchAndLaunchMetals(context, javaHome, activation))
     .catch((err) => {
       outputChannel.appendLine(err);
       showMissingJavaMessage();
+      activation.status.dispose();
     });
   commands.executeCommand("setContext", "metals:enabled", true);
+  setHasBuildTool();
+}
+
+function setHasBuildTool(): void {
+  let allPromises = [
+    "build.sbt",
+    "build.sc",
+    "project/build.properties",
+    "**/scala/**",
+  ].map(async (glob) => {
+    const files = await workspace.findFiles(glob);
+    return files.length == 0;
+  });
+  Promise.all(allPromises).then((all) => {
+    if (
+      all.every((a) => {
+        return a;
+      })
+    ) {
+      commands.executeCommand("setContext", "metals:no-build-tool", true);
+    }
+  });
 }
```

Comment:
Absolutely optional proposal to rewrite this. It's easier to read for me this way, but it's clearly subjective:

```suggestion
function setHasBuildTool(): Promise<void> {
  const buildToolGlobs = [
    "build.sbt",
    "build.sc",
    "project/build.properties",
    "**/scala/**",
  ]
 
  return Promise.all(buildToolGlobs.map(workspace.findFiles)).then(matchResults => {
    if (matchResults.every(matchedFiles => matchedFiles == 0) {
      return commands.executeCommand("setContext", "metals:no-build-tool", true);
    }
  })
}
```

(in Scala this would be a `traverse`, obviously :P)

Reply:
Thanks! I wasn't sure how to write this properly in type script.

Reply:
I actually refactored the glob pattern as suggested by @kpbochenek and it seems it should be better now.